File: CPP/Petshop.cpp
/*
Saya Muhammad Hafidh Fadhilah dengan NIM 2305672 mengerjakan Latihan Modul 1 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/
#include <iostream>
#include <string>

using namespace std;

class Petshop{
    private:
        // Atribut untuk menyimpan informasi produk
        string ID;        // ID produk
        string nama;      // Nama produk
        string kategori;  // Kategori produk
        int harga;        // Harga produk

    public:
        // Konstruktor default (inisialisasi tanpa nilai awal)
        Petshop(){
            this->ID = "";
            this->nama = "";
            this->kategori = "";
            this->harga = 0;
        }

        // Konstruktor parameter ( inisialisasi dengan nilai tertentu)
        Petshop(string ID, string nama, string kategori, int harga){
            this->ID = ID;
            this->nama = nama;
            this->kategori = kategori;
            this->harga = harga;
        }

        // Setter atribut
        void set_ID(string ID) {this->ID = ID;}
        void set_nama(string nama){this->nama = nama;}
        void set_kategori(string kategori){ this->kategori = kategori;}
        void set_harga(int harga){this->harga= harga;}
        
        // Getter atribut
        string get_ID(){return this->ID;}
        string get_nama(){return this->nama;}
        string get_kategori(){return this->kategori;}
        int get_harga(){return this->harga;}
        
        // Metode untuk menampilkan data dalam bentuk tabel
        void tampil(){
            // Garis pemisah tabel
            for(size_t i = 0; i < 80; i++) cout << "-";
            cout << '\n';

            // Menampilkan ID
            cout << this->ID;
            for (size_t i = this->ID.length();i < 5;i++) cout << " ";
            cout << " | ";

            // Menampilkan Nama
            cout << this->nama;
            for(size_t i = this->nama.length();i < 35;i++) cout << " ";
            cout << " | ";

            // Menampilkan Kategori
            cout << this->kategori;
            for(size_t i = this->kategori.length();i < 25;i++) cout << " ";
            cout << " | ";

            // Menampilkan Harga
            cout << this->harga << '\n';
        }

        // Destructor 
        ~Petshop() {}
};

File: Java/Petshop.java
public class Petshop{
    private String ID;
    private String nama;
    private String kategori;
    private int harga;

    // Constructor
    public Petshop(){
        this.ID="";
        this.nama="";
        this.kategori="";
        this.harga=0;
    }

    // Constructor dengan parameter
    public Petshop(String ID, String nama, String kategori, int harga){
        this.ID = ID;
        this.nama = nama;
        this.kategori = kategori;
        this.harga = harga;
    }

    // Setter
    public void set_ID(String ID) {this.ID = ID;}
    public void set_nama(String nama){this.nama = nama;}
    public void set_kategori(String kategori){this.kategori = kategori;}
    public void set_harga(int harga){this.harga = harga;}

    // Getter
    public String get_ID(){ return this.ID;}
    public String get_nama(){return this.nama;}
    public String get_kategori(){return this.kategori;}
    public int get_harga(){return this.harga;}

    // Menampilkan data
    public void tampil(){
        // Garis pemisah tabel
        System.out.println("-".repeat(80));

        // Menampilkan ID
        System.out.print(this.ID);
        System.out.print(" ".repeat(5 - ID.length()) + " | ");

        // Menampilkan Nama
        System.out.print(this.nama);
        System.out.print(" ".repeat(35 - nama.length()) + " | ");

        // Menampilkan Kategori
        System.out.print(this.kategori);
        System.out.print(" ".repeat(25 - kategori.length()) + " | ");

        // Menampilkan Harga
        System.out.println(this.harga);
    }
}